src/hotspot/share/classfile/javaClasses.cpp

```diff
@@ -2956,15 +2956,83 @@ void java_lang_StackTraceElement::decode(const methodHandle& method, int bci,
 }
 #endif // INCLUDE_JVMCI
 
+// java_lang_ClassFrameInfo
+
+int java_lang_ClassFrameInfo::_classOrMemberName_offset;
+int java_lang_ClassFrameInfo::_flags_offset;
+
+#define CLASSFRAMEINFO_FIELDS_DO(macro) \
+  macro(_classOrMemberName_offset, k, "classOrMemberName", object_signature,  false); \
+  macro(_flags_offset,             k, vmSymbols::flags_name(), int_signature, false)
+
+void java_lang_ClassFrameInfo::compute_offsets() {
+  InstanceKlass* k = vmClasses::ClassFrameInfo_klass();
+  CLASSFRAMEINFO_FIELDS_DO(FIELD_COMPUTE_OFFSET);
+}
+
+#if INCLUDE_CDS
+void java_lang_ClassFrameInfo::serialize_offsets(SerializeClosure* f) {
+  CLASSFRAMEINFO_FIELDS_DO(FIELD_SERIALIZE_OFFSET);
+}
+#endif
+
+static int get_flags(const methodHandle& m) {
+  int flags = (jushort)( m->access_flags().as_short() & JVM_RECOGNIZED_METHOD_MODIFIERS );
+  if (m->is_static()) {
+    flags |= java_lang_invoke_MemberName::MN_IS_METHOD;
+  } else if (m->is_initializer()) {
+    flags |= java_lang_invoke_MemberName::MN_IS_CONSTRUCTOR;
+  } else {
+    // "special" reflects that this is a direct call, not that it
+    // necessarily originates from an invokespecial. We can also do
+    // direct calls for private and/or final non-static methods.
+    flags |= java_lang_invoke_MemberName::MN_IS_METHOD;
+  }
```

Comment:
Both the is_static and the else block set java_lang_invoke_MemberName::MN_IS_METHOD.  Do we need to differentiate between the two cases or can they be collapsed?

Reply:
yes they can be collapsed.  I will update it.